src/Native/Unix/configure.cmake

```diff
@@ -407,6 +407,18 @@ check_function_exists(
     getpeereid
     HAVE_GETPEEREID)
 
+check_function_exists(
+    getdomainname
+    HAVE_GETDOMAINNAME)
+
+check_function_exists(
+    uname
+    HAVE_UNAME)
+
+if(NOT HAVE_GETDOMAINNAME AND NOT HAVE_UNAME)
+    message(FATAL_ERROR "Cannot find getdomainname nor uname on this platform.")
+endif()
```

Comment:
Nit: now that you added an else with ENOTSUP to the code, this check/error isn't strictly necessary.